lang/cs/Org.Apache.REEF.IMRU/API/IIMRUResultHandler.cs

```diff
@@ -0,0 +1,39 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using Org.Apache.REEF.IMRU.OnREEF.ResultHandler;
+using Org.Apache.REEF.Network.Group.Task.Impl;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.IMRU.API
+{
+    /// <summary>
+    /// Interface defining how to handle the result
+    /// </summary>
+    /// <typeparam name="TResult">Generic type</typeparam>
+    public interface IIMRUResultHandler<in TResult>
+    {
+        /// <summary>
+        /// Specifies how to handle the result. Write it to output/process it further/do nothing etc.
+        /// </summary>
+        /// <param name="result">The result of IMRU</param>
+        /// <param name="iterNo">Iteration number</param>
+        void ResultHandler(TResult result, int iterNo);
```

Comment:
Please document the exception behavior of this method. C# doesn't have a `throws` keyword, but it would still be nice to at least have documentation as to which exceptions implementations are supposed to throw and when. Also, is this interface to be implemented in a thread safe manner?